riocli/deployment/model.py

```diff
@@ -194,9 +193,14 @@ def create_object(self, client: Client, **kwargs) -> typing.Any:
         if 'features' in self.spec:
             if 'params' in self.spec.features and self.spec.features.params.enabled:
                 component_id = internal_component.componentId
+                disable_sync = self.spec.features.params.get('disableSync', False)
+
+                # Validate trees in the manifest with the ones available
+                # to avoid misconfigurations.
                 tree_names = self.spec.features.params.get('trees', [])
-                disable_sync = self.spec.features.params.get('disableSync',
-                                                             False)
+                available_trees = set(list_trees())
```

Comment:
suggestion: Consider caching the results or this will introduce one additional API call per deployment. The list output will most likely not change during the Apply cycle anyways.

Reply:
Done.